pims/image_sequence.py

```diff
@@ -248,87 +246,84 @@ class ImageSequence3D(ImageSequence):
         Image arrays will be converted to this datatype.
     as_grey : boolean, optional
         Not implemented for 3D images.
-    plugin : string
+    plugin : string, optional
         Passed on to skimage.io.imread if scikit-image is available.
         If scikit-image is not available, this will be ignored and a warning
         will be issued. Not available in combination with zipfiles.
-    tzc_identifiers : list of string, optional
-        3 strings preceding t, z, c indices. Default ['t', 'z', 'c'].
+    dim_identifiers : iterable of strings, optional
+        N strings preceding dimension indices. Default 'tzc'. x and y are not
+        allowed.
 
+    Attributes
+    ----------
+    dims : dict of FramesDimension objects
+        Dictionary of all available dimensions (excluding x and y)
+    ndim : int
+        Number of dimensions including x and y
+    sizes : dict of int
+        Dictionary with all dimension sizes
+    frame_shape : tuple of int
+        Shape of frames that will be returned by get_frame
+    frame_shape_2D : tuple of int
+        Size (y, x)
+    iterate : iterable of strings
+        This determines which dimensions will be iterated over by the
+        FramesSequence. The last element in will iterate fastest.
+    aggregate : iterable of strings
+        This determines which dimensions will be aggregated into one.
+        Frame. The dimensions in the ndarray that is returned by get_frame has
+        the same order as the order in this list.
     """
     def __init__(self, path_spec, process_func=None, dtype=None,
-                 as_grey=False, plugin=None, tzc_identifiers=None):
-        self.tzc_identifiers = tzc_identifiers
-        super(ImageSequence3D, self).__init__(path_spec, process_func,
+                 as_grey=False, plugin=None, dim_identifiers='tzc'):
+        if as_grey:
+            raise ValueError('As grey not supported for ND images')
+        self.dim_identifiers = dim_identifiers
+        super(ImageSequenceND, self).__init__(path_spec, process_func,
                                               dtype, as_grey, plugin)
+        if 't' in self.dims:
+            self.iterate = 't'  # iterate over t
+        if 'z' in self.dims:
+            self.aggregate = 'z'  # return z-stacks
 
     def _get_files(self, path_spec):
-        super(ImageSequence3D, self)._get_files(path_spec)
-        self._toc = np.array([filename_to_tzc(f, self.tzc_identifiers) \
+        super(ImageSequenceND, self)._get_files(path_spec)
```

Comment:
I have been working on a python3 only project, the better super support is great (`super().foo` works as expected)